### PR TITLE
Remove Python 2 dependency.

### PR DESCRIPTION
**Pull request by arokem**
Implements as discussed in #1731 

Also, https://github.com/travis-ci/travis-ci/issues/9815 seems to now be resolved.

### CODE REVIEW

**Comment by arokem** [comment]
These errors: https://travis-ci.org/nipy/dipy/jobs/506703626#L2364

Are addressed in nibabel (https://github.com/nipy/nibabel/pull/719)

I am not sure why they appear only on the machine with coverage turned on, though.

**Comment by arokem** [comment]
We should wait for a nibabel release to rebase this :-)

**Comment by effigies** [comment]
Coming up on Monday. :-)

**Comment by arokem** [comment]
Now, that it's Monday and Nibabel is released (👏 🎉 🍾), rebased to see what happens.

**Comment by effigies** [comment]
Well, it was *a* Monday. Thanks for your patience.

**Comment by arokem** [comment]
It's always a Monday somewhere, or something like that ;)

Thank *you* for all the hard work!

**Comment by skoudoro** [comment]
should we use this PR to reconsider our minimal version as discussed [here](https://github.com/nipy/dipy/pull/1708#issuecomment-450514806)?

**Comment by arokem** [comment]
Yeah. Good idea.

**Comment by arokem** [comment]
There is a numpy wheel for 3.7 on pypi: https://pypi.org/project/numpy/#files

**Comment by skoudoro** [comment]
> There is a numpy wheel for 3.7 on pypi: https://pypi.org/project/numpy/#files

Not for our minimal version Numpy 1.8.2 as you can see [here](https://pypi.org/project/numpy/1.8.2/#files)

**Comment by arokem** [comment]
Oh yeah - gotcha. 

Should we bump up our minimal numpy, so we can guarantee support for 3.7? Or provide support for 3.7 for only some versions of numpy? 

I would vote for the latter

File: setup_egg.py [hunk 1]
```diff
@@ -4,6 +4,6 @@
 """Wrapper to run setup.py using setuptools."""
 
 if __name__ == '__main__':
-    execfile('setup.py', dict(__name__='__main__',
-                              __file__='setup.py', # needed in setup.py
-                              force_setuptools=True))
+    exec('setup.py', dict(__name__='__main__',
```
[the remark this answers is not in the document]
Comment (skoudoro) [reply setup_egg.py]: Can you update this? `exec` and `execfile` are not called the same way finally. I should have check...

**Comment by skoudoro** [comment]
> I would vote for the latter

I will vote for that too if you explain to me how do you manage it? 😄

**Comment by arokem** [comment]
[This](https://github.com/nipy/dipy/pull/1775/commits/c3f44ad5c37fb759ddaf2e6bbd9f1dfddbb5e10a) is what I was thinking for the CI. 
 
For the requirements, I think that we would have to do something like what Adam did [here](https://github.com/richford/cloudknot/blob/master/cloudknot/version.py#L79) in Cloudknot.

**Comment by skoudoro** [comment]
Ok, I see, Nice! Thank you for that!

**Comment by skoudoro** [comment]
During your `force-pushed` it seems you have deleted [this](https://github.com/arokem/dipy/pull/32/files#diff-2d2616242db54458268e23e05ae71666R80) and [this](https://github.com/arokem/dipy/pull/32/files#diff-b4ef698db8ca845e5845c4618278f29aR2). Can put them back?

Thanks!

**Comment by arokem** [comment]
Whoopsie. I think that your change to setup_egg is superseded by https://github.com/nipy/dipy/pull/1775/commits/ddfa2f4e9c1ca80e4b85017fa8af4c2783da2007, but I cherry-picked back your dipy/info.py, because I am a git wizard.

**Comment by arokem** [comment]
Looks like Travis is all 💚 now! Let's see what Appveyor says.

**Comment by skoudoro** [comment]
😍, look forward to press "merge"

**Comment by skoudoro** [comment]
moving to python 3 only! Thanks @arokem! merging

**Comment by arokem** [comment]
🎉 🍾 🌟

**Pull request merged**